src/FailureDetectorConfig.md

```diff
@@ -85,3 +88,59 @@ config.setProperty("hazelcast.heartbeat.phiaccrual.failuredetector.sample.size",
+isn't met, Hazelcast will fail to start.
+
+Below a summary table of all possible combinations of the ping failure detector.
+
+| ICMP  	| Parallel 	| Fail-Fast 	| Description                                                                                                                                                                                                                 	| Linux                                                                                                  	| Windows                       	| macOS                                                                	|
+|-------	|----------	|-----------	|-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------	|--------------------------------------------------------------------------------------------------------	|-------------------------------	|----------------------------------------------------------------------	|
+| false 	| false    	| false     	| Completely disabled                                                                                                                                                                                                         	| N/A                                                                                                    	| N/A                           	| N/A                                                                  	|
+| true  	| false    	| false     	| Legacy ping mode. Pings nodes up-to 5 times after a pre-defined no heartbeat period has passed. If all 5 attempts fail, node gets suspected. It relies on the configured failure detector (OSI Layer 7) before it kicks-in. 	| Supported  ICMP Echo if available - Falls back on TCP Echo on port 7                                   	| Supported  TCP Echo on port 7 	|  Supported ICMP Echo if available - Falls back on TCP Echo on port 7 	|
+| true  	| true     	| false     	| Parallel ping detector, works in parallel with the configured failure detector. Checks periodically if nodes are live (OSI Layer 3), and suspects them immediately, regardless of the other detectors.                      	| Supported  ICMP Echo if available - Falls back on TCP Echo on port 7                                   	| Supported  TCP Echo on port 7 	| Supported  ICMP Echo if available - Falls back on TCP Echo on port 7 	|
+| true  	| true     	| true      	| Parallel ping detector, works in parallel with the configured failure detector. Checks periodically if nodes are live (OSI Layer 3), and suspects them immediately, regardless of the other detectors.                      	| Supported - Requires OS Configuration  Enforcing ICMP Echo if available - No start up if not available 	| Not Supported                 	| Not Supported - Requires root priviledges                            	|
```

Comment:
Isn't `| true | false | true` a valid option too? You don't enable parallel mode but want to use old ping fd with fail fast option.

Reply:
actually, it could be, but it isn't. this code path is only available for parallel-mode only. I guess we could improve it further, but this will need code changes & testing and I wouldn't want to risk this last minute. I can include for 3.10

Reply:
fair enough.